tests/kserve_models_web_application_test.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+set -euxo pipefail
+
+# KServe Models Web Application Test Script
+# Tests the models-web-application API functionality with kubectl-deployed InferenceServices
+
+KF_PROFILE=${1:-kubeflow-user-example-com}
+TOKEN="$(kubectl -n $KF_PROFILE create token default-editor)"
+BASE_URL="localhost:8080/kserve-endpoints"
+
+cat <<EOF | kubectl apply -f -
+apiVersion: "serving.kserve.io/v1beta1"
+kind: "InferenceService"
+metadata:
+  name: "sklearn-iris-private"
+  namespace: ${KF_PROFILE}
+spec:
+  predictor:
+    sklearn:
+      storageUri: "gs://kfserving-examples/models/sklearn/1.0/model"
+      resources:
+        requests:
+          cpu: "50m"
+          memory: "128Mi"
+        limits:
+          cpu: "100m"
+          memory: "256Mi"
+EOF
+
+kubectl wait --for=condition=Ready inferenceservice/sklearn-iris-private -n ${KF_PROFILE} --timeout=300s
+kubectl get inferenceservice sklearn-iris-private -n ${KF_PROFILE}
+
+# Get XSRF token for API calls
+curl -s "http://${BASE_URL}/" \
+  -H "Authorization: Bearer ${TOKEN}" \
+  -v -c /tmp/kserve_xcrf.txt 2>&1 | grep -i "set-cookie"
+XSRFTOKEN=$(grep XSRF-TOKEN /tmp/kserve_xcrf.txt | awk '{print $NF}')
+
+RESPONSE=$(curl -s --fail-with-body \
+  "${BASE_URL}/api/namespaces/${KF_PROFILE}/inferenceservices" \
+  -H "Authorization: Bearer ${TOKEN}" \
+  -H "X-XSRF-TOKEN: ${XSRFTOKEN}" \
+  -H "Cookie: XSRF-TOKEN=${XSRFTOKEN}")
+
+echo "API Response:"; head -c 500 <<< "$RESPONSE"; echo
+
+echo "$RESPONSE" | grep -q "sklearn-iris-private" || exit 1
+kubectl get inferenceservice sklearn-iris-private -n ${KF_PROFILE} || exit 1
+READY=$(kubectl get isvc sklearn-iris-private -n ${KF_PROFILE} -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}')
+[[ "$READY" == "True" ]] || {
+  echo "FAILURE: InferenceService Ready status is: $READY"
+  exit 1
+}
+
+TODO verify that it fails for security without the token.
```

Comment:
Please add a minimal and elegant test that succeeds if you get an error with a wrong token or no token at all

Reply:
See for example https://github.com/LogicalGuy77/manifests-logical/blob/7d2c8629008f14a67528d19b6f25ca9e040257b6/.github/workflows/pipeline_test.yaml#L92

Reply:
Or https://github.com/kubeflow/manifests/blob/2937b109463371c2dc15ed29f2cc75485016fda7/tests/kserve_test.sh#L90